### DISCUSSION TITLE
Is it possible to get nested tables of data in a model then we will apply searching and sorting. Is it a good way ??

### DISCUSSION THREAD

**Discussion by KhushbuAsrani08** [prisma/prisma#18232] [Q&A] [answered]
**![image](https://user-images.githubusercontent.com/125308219/223135591-2f89690d-f26f-4d40-b1de-73b4607923ae.png)**

In the above image of Owner and No.of users column wants to apply sorting and searching from the server side (in prisma query only).
This is my prisma query : 

 ```
query: (paginateArgs) =>
          db.project.findMany({
            orderBy: [{ [columnName]: columnOrder }],
            where: {
              id: { in: [...projectIds] },
              deleted_at: { gte: new Date() },
              organization_id: session.orgId,
              OR : [{name: { contains : !isNil(searchVal)? searchVal : ""}},
               {project_status_type_id : {contains : !isNil(searchVal)? searchVal : ""}},
               {project_users : {some : {user : {name : {contains : !isNil(searchVal) ? searchVal : ""}}}}}
            ]
            },
            include: {
              project_users: { where: { is_enabled: true } , include: { user: true } },
              project_invitation_pending_users: true,
            },
            ...paginateArgs,
           
          }),
```
        
Project table in relation with project_users and project_users table in relation with user table.
Owner and no. of users of data in manipulated at frontend.

**Comment by ludralph** [comment, marked as answer]
Hi @KhushbuAsrani08 👋 

it is possible to have nested tables of data in a model and apply searching and sorting on it using Prisma queries. Prisma provides powerful filtering and sorting capabilities that allow you to search and sort data based on different criteria.

In your Prisma query, you use the include keyword to fetch related data from the project_users table and the project_invitation_pending_users table. You can also use the orderBy keyword to specify the sorting order based on a particular column, as you have done in your query. To search for nested data, you can use the contains filter along with the some operator to search for a specific value in an array of related data. For example, in your query, you are searching for a user name that contains a specific value using the following filter:

```js
{project_users : {some : {user : {name : {contains : !isNil(searchVal) ? searchVal : ""}}}}}
```

This filter searches for the specified value in the name column of the user table, which is a nested table inside the project_users table.

Overall, it is a good way to use nested tables and apply searching and sorting on them, allowing you to organize and manipulate data more efficiently.

**Comment by KhushbuAsrani08** [comment]
Got an answer
